tika-core/src/main/java/org/apache/tika/detect/DefaultEncodingDetector.java

```diff
@@ -37,17 +40,43 @@
  */
 public class DefaultEncodingDetector extends CompositeEncodingDetector {
 
+    private static final long serialVersionUID = 1L;
+
+    private ServiceLoader loader;
+
+    private List<EncodingDetector> dynamicDetectors;
+    private boolean loadDynamicServices;
+
     public DefaultEncodingDetector() {
-        this(new ServiceLoader(DefaultEncodingDetector.class.getClassLoader()));
+        this(false);
+    }
+
+    public DefaultEncodingDetector(boolean loadDynamicServices) {
+        this(new ServiceLoader(DefaultEncodingDetector.class.getClassLoader(), loadDynamicServices));
+        this.loadDynamicServices = loadDynamicServices;
     }
 
     public DefaultEncodingDetector(ServiceLoader loader) {
-        super(loader.loadServiceProviders(EncodingDetector.class));
+        super(loader.loadStaticServiceProviders(EncodingDetector.class));
+        dynamicDetectors = loader.loadDynamicServiceProviders(EncodingDetector.class);
+        this.loader = loader;
     }
 
     public DefaultEncodingDetector(ServiceLoader loader,
                                    Collection<Class<? extends EncodingDetector>> excludeEncodingDetectors) {
-        super(loader.loadServiceProviders(EncodingDetector.class), excludeEncodingDetectors);
+        super(loader.loadStaticServiceProviders(EncodingDetector.class), excludeEncodingDetectors);
```

Comment:
This is going to change the old behavior. The old behavior might have loaded static and dynamic providers. Now only static providers are loaded and filtered.